commit 95aa474fa6a14248d37022b1894b27b28c6173cd
Author: Albert Astals Cid <aacid@kde.org>
Date:   Wed Mar 6 20:14:25 2013 +0100

    Give mFakeProcess a parent (itself)
    
    We need this since all processes need to have a parent because of constructs like
    
            do {
                p = p->parent;
                p->numChildren--;
            } while (p->pid!= -1);
    
    If p is the mFakeProcess it would crash (as demonstrated in processtest.cpp)
    now it properly ends the loop and the test passes
    
    CCMAIL: johnflux@gmail.com

diff --git a/libs/ksysguard/processcore/processes.cpp b/libs/ksysguard/processcore/processes.cpp
index 14da8cb..481c680 100644
--- a/libs/ksysguard/processcore/processes.cpp
+++ b/libs/ksysguard/processcore/processes.cpp
@@ -48,6 +48,7 @@ namespace KSysGuard
   {
     public:
       Private(Processes *q_ptr) {
+        mFakeProcess.parent = &mFakeProcess;
         mAbstractProcesses = 0;
         mHistoricProcesses = 0;
         mIsLocalHost = true;
